static/panes/haskellcmm-view.ts

```diff
@@ -0,0 +1,118 @@
+// POSSIBILITY OF SUCH DAMAGE.
+
+import _ from 'underscore';
+import * as monaco from 'monaco-editor';
+import {Container} from 'golden-layout';
+
+import {MonacoPane} from './pane';
+import {MonacoPaneState} from './pane.interfaces';
+import {HaskellCmmState} from './haskellcmm-view.interfaces';
+
+import {ga} from '../analytics';
+import {extendConfig} from '../monaco-config';
+import {Hub} from '../hub';
+
+export class HaskellCmm extends MonacoPane<monaco.editor.IStandaloneCodeEditor, HaskellCmmState> {
+    constructor(hub: Hub, container: Container, state: HaskellCmmState & MonacoPaneState) {
+        super(hub, container, state);
+        if (state.haskellCmmOutput) {
+            this.showHaskellCmmResults(state.haskellCmmOutput);
+        }
+    }
+
+    override getInitialHTML(): string {
+        return $('#haskellCmm').html();
+    }
+
+    override createEditor(editorRoot: HTMLElement): monaco.editor.IStandaloneCodeEditor {
+        return monaco.editor.create(
+            editorRoot,
+            extendConfig({
+                language: 'haskell',
+                readOnly: true,
+                glyphMargin: true,
+                lineNumbersMinChars: 3,
+            })
+        );
+    }
+
+    override registerOpeningAnalyticsEvent(): void {
+        ga.proxy('send', {
+            hitType: 'event',
+            eventCategory: 'OpenViewPane',
+            eventAction: 'HaskellCmm',
+        });
+    }
+
+    override getDefaultPaneName(): string {
+        return 'GHC Cmm viewer';
+    }
+
+    override registerCallbacks(): void {
+        const throttleFunction = _.throttle(event => this.onDidChangeCursorSelection(event), 500);
+        this.editor.onDidChangeCursorSelection(event => throttleFunction(event));
+        this.eventHub.emit('haskellCmmViewOpened', this.compilerInfo.compilerId);
+        this.eventHub.emit('requestSettings');
+    }
+
+    override onCompileResult(compilerId: number, compiler: any, result: any): void {
+        if (this.compilerInfo.compilerId !== compilerId) return;
+        if (result.hasHaskellCmmOutput) {
+            this.showHaskellCmmResults(result.haskellCmmOutput);
+        } else if (compiler.supportsHaskellCmmView) {
+            this.showHaskellCmmResults([{text: '<No output>'}]);
+        }
+    }
+
+    override onCompiler(compilerId: number, compiler: any, options: any, editorId?: number, treeId?: number): void {
+        if (this.compilerInfo.compilerId === compilerId) {
+            this.compilerInfo.compilerName = compiler ? compiler.name : '';
+            this.compilerInfo.editorId = editorId;
+            this.compilerInfo.treeId = treeId;
+            this.updateTitle();
+            if (compiler && !compiler.supportsHaskellCmmView) {
+                this.showHaskellCmmResults([{text: '<GHC Cmm output is not supported for this compiler>'}]);
+            }
+        }
+    }
+
+    showHaskellCmmResults(result: any[]): void {
```

Comment:
While the type is probably to restrictive, this could be `result: Record<'text', string>[]`